plugins/@grouparoo/google-sheets/__tests__/cli/integration.ts

```diff
@@ -0,0 +1,59 @@
+import path from "path";
+import { CLISpecHelper } from "@grouparoo/spec-helper";
+
+const pluginName = "@grouparoo/google-sheets";
+const pluginPath = path.join(__dirname, "..", "..");
+
+describe("google-sheets/CLI", () => {
+  const { projectPath, runCliCommand } = CLISpecHelper.prepareForCLITest(
+    pluginName,
+    pluginPath
+  );
+
+  test("the plugin appears in the generate list", async () => {
+    const { stdout, exitCode } = await runCliCommand(
+      "generate --list".split(" ")
+    );
+
+    expect(exitCode).toBe(0);
+    expect(stdout).toContain("google-sheets:app");
+    expect(stdout).toContain("google-sheets:property");
+    expect(stdout).toContain("google-sheets:source");
+  });
+
+  test("an app can be generated", async () => {
+    const { exitCode } = await runCliCommand(
+      "generate google-sheets:app app".split(" ")
+    );
+
+    expect(exitCode).toBe(0);
+  });
+
+  test("a source can be generated", async () => {
+    const { exitCode } = await runCliCommand(
+      "generate google-sheets:source source --parent app".split(" ")
+    );
+
+    expect(exitCode).toBe(0);
+    CLISpecHelper.enableBootstrappedProperty(projectPath, "source");
+  });
+
+  test("a property can be generated", async () => {
+    const { exitCode } = await runCliCommand(
+      "generate google-sheets:property property property --parent source".split(
+        " "
+      )
+    );
+
+    expect(exitCode).toBe(0);
+  });
+
+  test("the config files can be locally validated", async () => {
+    const { stdout, exitCode } = await runCliCommand(
+      "validate --local".split(" ")
+    );
+
+    expect(exitCode).toBe(0);
+    expect(stdout).toContain("Validation succeeded");
+  });
```

Comment:
This test works because we can `validate --local`.  This will skip testing against "real" data, meaning we don't need to use real usernames and passwords, but it will test that the plugins, sources, and properties are configured correctly, have the same name, and that the options required are present in the template.

This test confirms the bug has been fixed.

Reply:
Let's think about a way to do this for all plugins and templates in another PR sometime.

Reply:
Already updated https://www.pivotaltracker.com/story/show/177463643!